python/paddle/v2/framework/tests/gradient_checker.py

```diff
@@ -76,6 +95,131 @@ def product(dim):
     return gradient_flat.reshape(tensor_to_check.get_dims())
 
 
+class GradientChecker(unittest.TestCase):
+    def check_grad(self,
+                   forward_op,
+                   input_vars,
+                   inputs_to_check,
+                   output_name,
+                   no_grad_set=set(),
+                   only_cpu=False,
+                   rtol=0.005,
+                   atol=0.05):
+        """
+        :param forward_op: used to create backward_op
+        :param input_vars: numpy value of input variable. The following
+            computation will use these variables.
+        :param inputs_to_check: inputs var names that should check gradient.
+        :param output_name: output name that used to
+        :param rtol: The relative tolerance parameter.
+        :param atol: The absolute tolerance parameter.
+        :param no_grad_set: used when create backward ops
+        :param only_cpu: only compute and check gradient on cpu kernel.
+        :return:
+        """
+        out_names = filter(lambda name: name != "@TEMP@", forward_op.outputs())
```

Comment:
Not only `@TEMP@` in forward_op is temporary. use forward_op.attr("temporary_index") to get no temporary output.